srml/eth-relay/src/lib.rs

```diff
@@ -157,8 +158,14 @@ decl_event! {
 
 impl<T: Trait> Module<T> {
 	// TOOD: what is the total difficulty for genesis/begin header
-	pub fn init_genesis_header(header: &EthHeader, genesis_difficulty: u64) {
+	pub fn init_genesis_header(header: &EthHeader, genesis_difficulty: u64) -> result::Result<(), &'static str> {
 		let header_hash = header.hash();
+
+		ensure!(
+			header_hash == header.re_compute_hash(),
+			"Header Hash Check Failed - Hash NOT MATCH",
```

Comment:
I think Header Hash - MISMATCHED is better.

Could you please replace all NOT MATCHED with MISMATCHED?